database/sql/src/main/scala/cromwell/database/sql/MetadataSqlDatabase.scala

```diff
@@ -139,4 +139,6 @@ trait MetadataSqlDatabase extends SqlDatabase {
                              metadataArchiveStatus: Set[Option[String]],
                              includeSubworkflows: Boolean)
                              (implicit ec: ExecutionContext): Future[Int]
+
+  def deleteNonLabelMetadataForWorkflow(rootWorkflowId: String)(implicit ec: ExecutionContext): Future[Int]
```

Comment:
The signature here would additionally take in the `statusesOkToDelete: Iterable[String]`